File: Java/src/parser/ast/Variable.java
package parser.ast;

import lexer.Token;

/**
 * Author : lihuichuan
 * Time   : 2020/12/3
 */
public class Variable extends Factor{
    public Variable(Token token) {
        super(token);
        this.type = ASTNodeTypes.VARIABLE;
    }


}

File: JS/src/parser/ast/Variable.js
const ASTNodeTypes = require('./ASTNodeTypes')
const Factor = require('./Factor')
class Variable extends Factor{

  
    constructor(token){
        super(token)
        this.type = ASTNodeTypes.VARIABLE
        this.typeLexeme = null
    }


    setTypeLexeme(lexeme){
        this.typeLexeme = lexeme
    }

    getTypeLexeme(){
        return this.typeLexeme
    }


    
}

module.exports = Variable
